### PR TITLE
Pez/vsix+fs testing flakyness

### LINKED ISSUE #154: We should test the VSIX package in addition to from the filesystem
Today we run our integrations tests with the extension as built in the filesystem. Even though it is a release build, there is enough that can go wrong with the packaged VSIX to warrant a test of that too.

### PR DESCRIPTION
**Pull request by PEZ**
* Created a dummy extension to feed the vscode-test launcher when we are testing the extension from the filesystem
* Installing extensions and user data in the local VS Code Insiders (not the Machine/global install as before). Should make it safer and more stable to run the tests locally.

* Fixes #154 

- [x] I have read the [developer documentation](https://github.com/BetterThanTomorrow/joyride/blob/master/CONTRIBUTE.md).

- [x] This PR corresponds to an [issue with a clear problem statement](https://github.com/babashka/babashka/blob/master/doc/dev.md#start-with-an-issue-before-writing-code).

- [x] This PR contains a [test](https://github.com/BetterThanTomorrow/joyride/tree/master/vscode-test-runner/workspace-1/.joyride/src/integration_test) to prevent against future regressions

- [x] I have updated the **\[Unreleased\]** section of the [CHANGELOG.md](https://github.com/BetterThanTomorrow/joyride/blob/master/CHANGELOG.md) file with a link to the addressed issue.

### CODE REVIEW

**Comment by borkdude** [comment]
@PEZ I'll re-run this a few times to double-check on the flakiness :-D

File: .circleci/config.yml [hunk 1]
```diff
@@ -296,7 +325,15 @@ workflows:
...
       #        only: /^v\d+\.\d+\.\d+-?.*/
-      - test-integration:
+      - test-integration-fs:
+          requires:
+            - build
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /^v\d+\.\d+\.\d+-?.*/
```
Comment (borkdude) [inline .circleci/config.yml:335]: Why is this?

**Comment by borkdude** [comment]
Other than my question: LGTM!

File: .circleci/config.yml (hunk 1 above)
Comment (PEZ) [reply .circleci/config.yml:335]: Let's see... This is part of the publish-release pipeline and we only want to run this if we're on a version tag. We allow for pre-releases in waiting for semver pre-releases to be supported by the marketplace. Which maybe will never happen.

* https://github.com/microsoft/vscode-vsce/issues/148
* https://github.com/microsoft/vscode/issues/15756#issuecomment-916728348

It's copied over from Calva where more of the pre-release publishing is prepared for. We could consider cleaning it away from here.

**Pull request merged**